### PR TITLE
fix(input, input-number): support setting value property to Infinity

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #7866

## Summary

- Setting the value of `calcite-input` and `calcite-input-number` to `Infinity` or `-Infinity` no longer causes errors
- `Infinity` or `-Infinity` will be displayed in the input.
- Entering additional characters is prevented, and pressing Backspace or Delete will clear the value.

### CODE REVIEW

**Comment by benelan** [comment]
We might want to wait for @jcfranco on this one. I'm not sure if he'll want us to translate `Infinity` or not. FWIW we don't translate the `e` in exponential numbers.

File: packages/calcite-components/src/components/input/input.e2e.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/input/input.e2e.ts:1946]: ✨🏆✨

**Review by jcfranco — APPROVED** [review]
♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️
♾️♾️😎😎😎♾️😎♾️♾️♾️♾️♾️😎♾️😎😎😎😎♾️😎😎😎😎♾️😎😎😎😎😎♾️😎♾️
♾️😎♾️♾️♾️♾️😎♾️♾️♾️♾️♾️😎♾️😎♾️♾️♾️♾️😎♾️♾️♾️♾️♾️♾️😎♾️♾️♾️😎♾️
♾️♾️😎😎♾️♾️😎♾️♾️😎♾️♾️😎♾️😎😎😎♾️♾️😎😎😎♾️♾️♾️♾️😎♾️♾️♾️😎♾️
♾️♾️♾️♾️😎♾️😎♾️😎♾️😎♾️😎♾️😎♾️♾️♾️♾️😎♾️♾️♾️♾️♾️♾️😎♾️♾️♾️♾️♾️
♾️😎😎😎♾️♾️♾️😎♾️♾️♾️😎♾️♾️😎😎😎😎♾️😎😎😎😎♾️♾️♾️😎♾️♾️♾️😎♾️
♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️♾️

Since this is displaying a special value, could you add a screenshot test for all infinity cases?

> We might want to wait for @jcfranco on this one. I'm not sure if he'll want us to translate Infinity or not. FWIW we don't translate the e in exponential numbers.

I think we can proceed w/ `Infinity` for the time being as, IMO, it's an edge case. From a design and localization-perspective, would displaying `∞` vs `Infinity` (value setting/event emitting would still use `Infinity`) be a good alternative that doesn't need translation? cc @SkyeSeitz @ashetland @annierm18

**Comment by benelan** [comment]
@jcfranco is this okay to merge or should we wait for i18n feedback from @annierm18 regarding:

> From a design and localization-perspective, would displaying `∞` vs `Infinity` (value setting/event emitting would still use `Infinity`) be a good alternative that doesn't need translation?

**Comment by ashetland** [comment]
Don't have a strong opinion on this one. `Infinity` _might_ be more clear, but defer to @SkyeSeitz and @annierm18.

**Comment by SkyeSeitz** [comment]
I don't see a large issue with either option so I don't have a strong opinion here either, but the symbol does seem to read clear enough and may worth it if translation is an issue, especially if we have other similar symbols in use. 
image

**Comment by jcfranco** [comment]
> @jcfranco is this okay to merge

Yus. Sorry for not being explicit about the `∞` suggestion being a follow-up.

File: packages/calcite-components/src/components/input/input.tsx
Comment (benelan) [inline packages/calcite-components/src/components/input/input.tsx:131]: @jcfranco do you know why this keeps ending up on a new line? Lol

Comment (jcfranco) [reply packages/calcite-components/src/components/input/input.tsx:131]: It looks like it's the result of my habit of placing decorators above of their target and https://github.com/stencil-community/stencil-eslint/blob/main/docs/decorators-style.md not auto fixing for some reason. Does it auto fix this for you when explicitly running the linter?

**Pull request merged**
